Assignment2_q1_test.py

```diff
@@ -0,0 +1,51 @@
+"""Tests for question 1 - print ancestors."""
+
+import Assignment2_q1
+import unittest
+
+
+class Q1Test(unittest.TestCase):
+
+    def testEmptyCase(self):
+        empty_tree = Assignment2_q1.BinaryTree()
+        with self.assertRaises(IndexError):
+            Assignment2_q1.print_ancestors(empty_tree, 0)
+
+    def testKnownCases(self):
+        # Create given tree example
+        node_1 = Assignment2_q1.TreeNode(1)
+        node_5 = Assignment2_q1.TreeNode(5)
+        node_3 = Assignment2_q1.TreeNode(3, node_1, node_5)
+        node_14 = Assignment2_q1.TreeNode(14)
+        node_9 = Assignment2_q1.TreeNode(9, node_3, node_14)
+        node_19 = Assignment2_q1.TreeNode(19)
+        node_18 = Assignment2_q1.TreeNode(18, None, node_19)
+        node_16 = Assignment2_q1.TreeNode(16, node_9, node_18)
+        tree = Assignment2_q1.BinaryTree(node_16)
+        self.assertEquals([5,3,9,16], Assignment2_q1.print_ancestors(tree, 5))
+        self.assertEquals([1,3,9,16], Assignment2_q1.print_ancestors(tree, 1))
+        self.assertEquals([3,9,16], Assignment2_q1.print_ancestors(tree, 3))
+        self.assertEquals([14,9,16], Assignment2_q1.print_ancestors(tree, 14))
+        self.assertEquals([19,18,16], Assignment2_q1.print_ancestors(tree, 19))
+        self.assertEquals([9,16], Assignment2_q1.print_ancestors(tree, 9))
+        self.assertEquals([18,16], Assignment2_q1.print_ancestors(tree, 18))
+        self.assertEquals([16], Assignment2_q1.print_ancestors(tree, 16))
+
+    def testErrorCases(self):
+        # Create given tree example
+        node_1 = Assignment2_q1.TreeNode(1)
```

Comment:
I've added seUp() method in both tests 👍 I haven't used unittests before the CodeU program and thus I didn't know about setUp() method, thanks a lot for your advice!